stablehlo/reference/tests/add.mlir

```diff
@@ -0,0 +1,270 @@
+// RUN: stablehlo-interpreter-runner -split-input-file %s | FileCheck %s
```

Comment:
Let's move these tests under `tests/`? This way we won't need to have another `lit.cfg.py` and `lit.site.cfg.py.in`.